Carbon.Domain.EntityFrameworkCore.Extensions/EFExtensions.cs

```diff
@@ -275,19 +275,21 @@ public static PagedList<T> ToPagedListEntityFilteredWithSolution<T, U>(
             var filterContainsData = filter != null && filter.Any();
 
             var query = relationEntities
-                .Where(k => !filterContainsData || k.Relation == null || filter.Contains(k.Relation.SolutionId))
-                .GroupBy(k => k.Entity, k => k.Relation,
-                    (key, g) => new EFExtensions.RelationEntity<T, U> { Entity = key, Relations = g.ToList() })
-                .Select(k => k.Entity)
-                .AsQueryable();
+                .Where(k => !filterContainsData || k.Relation == null || filter.Contains(k.Relation.SolutionId));
+
 
             var filteredQuery = query.OrderBy(orderables);
```

Comment:
@anil-eladag "orderables" must be optional for the query performance. So if can we check orderables is {Count:>0}, it will be useful.